internal/proxy/proxy_test.go

```diff
@@ -0,0 +1,170 @@
+package proxy_test
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/ubuntu-proxy-manager/internal/proxy"
+	"github.com/ubuntu/ubuntu-proxy-manager/internal/testutils"
+)
+
+func TestNew(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		http    string
+		https   string
+		ftp     string
+		socks   string
+		noProxy string
+		mode    string
+
+		wantErr bool
+	}{
+		"No options set":   {},
+		"HTTP option set":  {http: "http://example.com:8080"},
+		"Some options set": {http: "http://example.com:8080", https: "https://example.com:8080"},
+		"All options set":  {http: "http://example.com:8080", https: "https://example.com:8080", ftp: "ftp://example.com:8080", socks: "socks://example.com:8080", noProxy: "localhost,127.0.0.1", mode: "auto"},
+
+		// Error cases
+		"Error on unparsable URI for HTTP":          {http: "http://pro\x7Fy:3128", https: "https://valid.proxy", wantErr: true},
+		"Error on unparsable URI for HTTP (global)": {http: "http://pro\x7Fy:3128", wantErr: true},
+		"Error on unparsable URI for HTTPS":         {https: "http://pro\x7Fy:3128", wantErr: true},
+		"Error on unparsable URI for FTP":           {ftp: "http://pro\x7Fy:3128", wantErr: true},
+		"Error on unparsable URI for SOCKS":         {socks: "http://pro\x7Fy:3128", wantErr: true},
+		"Error on missing scheme":                   {socks: "example.com:8080", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		name := name
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			_, err := proxy.New(context.Background(), tc.http, tc.https, tc.ftp, tc.socks, tc.noProxy, tc.mode)
+			if tc.wantErr {
+				require.Error(t, err, "New should have failed but didn't")
+				return
+			}
+			require.NoError(t, err, "New failed but shouldn't have")
+		})
+	}
+}
+
+func TestApply(t *testing.T) {
+	t.Parallel()
+
+	envConfigPath := filepath.Join("etc", "environment.d", "99ubuntu-proxy.conf")
+
+	initialTime := time.Date(2001, 1, 1, 1, 1, 1, 1, time.UTC)
```

Comment:
or `Time.Unix(0)` or your birthday date :)

More seriously, we can stat the file after copying it, wdyt?

Reply:
The resolution is nanosecond based, and at least, on other projects, this works :)

Reply:
why? You can get current time after your crated your own file within the subtest itself?

Reply:
ah, you’re right, we are in Apply(), so we will have multiuple files :)

Ok, my only request then is to take a very identifiable date (unix 0) rather than a random one in 2001 (it was a bad year anyway :p)